frontend/apps/reader/modules/readertoc.lua

```diff
@@ -62,6 +62,7 @@ end
 
 function ReaderToc:onUpdateToc()
     self:resetToc()
+    self.ui:handleEvent(Event:new("TocUpdated"))
```

Comment:
resetToc() just clears the TOC data (so it's rebuild later when needed).
Don't you need it updated for when you catch TocUpdated (or so that that "TocUpdated" event is true to its name) ?
(But beware, rebuilding the TOC might be expensive, so that's why we delay it until needed.)

Reply:
Hmm... Maybe I should just change the name, because as I see it, when the TOC is reset, the footer needs to rebuild it to get the markers. So yes, I need the updated TOC, but I guess this need triggers the rebuild.

Reply:
Also please have a quick check that on loading a book (with cache or without), the TOC is only built once.
(I remember #3292 and https://github.com/koreader/koreader/issues/3261#issuecomment-333560902 - which are quite old - dunno if these thoughts are still relevant :)

Reply:
Right, I think there was one TOC too many which should be fixed now.